RecoMuon/MuonIdentification/plugins/MuonIdProducer.cc

```diff
@@ -799,15 +858,24 @@ void MuonIdProducer::fillMuonId(edm::Event& iEvent, const edm::EventSetup& iSetu
       }
       aMuon.setCalEnergy( muonEnergy );
    }
-   if ( ! fillMatching_ && ! aMuon.isTrackerMuon() && ! aMuon.isRPCMuon() ) return;
+
+   if (doME0_){
+     if ( ! fillMatching_ && ! aMuon.isTrackerMuon() && ! aMuon.isRPCMuon() && ! aMuon.isME0Muon() ) return;
```

Comment:
Lines 865-867 can be deleted by condensing lines 862 & 863:
`if ( ! fillMatching_ && ! aMuon.isTrackerMuon() && ! aMuon.isRPCMuon() && (! doME0_ || ! aMuon.isME0Muon()) )`